### PR TITLE
SQL Remodeling

### PR DESCRIPTION
**Pull request by clark800**

### CODE REVIEW

File: src/models/db/fulfillments.js [hunk 1]
```diff
@@ -22,17 +23,35 @@ function convertToPersistent (data) {
   return _.mapKeys(result, (value, key) => key.toUpperCase())
 }
 
+function * convertToIntegerTransferId (fulfillment, options) {
```
Comment (MatthewPhinney) [inline src/models/db/fulfillments.js]: the "IntegerTransferId" is the primary key, right?

Comment (clark800) [inline src/models/db/fulfillments.js]: Yeah, the primary key in the database

File: src/models/transfers.js [hunk 2]
```diff
@@ -438,6 +438,7 @@ function * setTransfer (externalTransfer, requestingUser) {
         previousCredits, 'credit')
     }
 
+    yield db.upsertTransfer(transfer, {transaction})
```
Comment (MatthewPhinney) [inline src/models/transfers.js]: Why is this needed?

File: src/sql/strong-oracle/create.sql [hunk 3]
```diff
@@ -3,255 +3,211 @@ CREATE SEQUENCE L_SEQ_ACCOUNT_PK
...
 CREATE INDEX L_XAK_TRANSFERS ON "L_TRANSFERS"
   ("TRANSFER_UUID" ASC);
-/
-
 ALTER TABLE "L_TRANSFERS" ADD CONSTRAINT L_AK_TRANSFERS UNIQUE
   ("TRANSFER_UUID");
-/
-
-ALTER TABLE "L_TRANSFERS" ADD (CONSTRAINT "FK_REJECTION_REASON_ID__TRANSF"
+CREATE BITMAP INDEX L_XIF_TRANSFERS_STATUS ON "L_TRANSFERS"
```
Comment (MatthewPhinney) [inline src/sql/strong-oracle/create.sql]: You are using a bitmap index because there are few distinct `status_id`s? Is this Abrahams' suggestion?

File: src/models/transfers.js (hunk 2 above)
Comment (clark800) [inline src/models/transfers.js]: The transfer has to exist before holds can be created; I'll add a comment for this

File: src/sql/strong-oracle/create.sql (hunk 3 above)
Comment (clark800) [inline src/sql/strong-oracle/create.sql]: Yeah, this is from Abraham's recommendation

File: src/sql/strong-oracle/disable.sql [hunk 4]
```diff
@@ -0,0 +1,119 @@
+
```
Comment (MatthewPhinney) [inline src/sql/strong-oracle/disable.sql]: Is there a way to combine this into a loop over all the tables?

File: src/sql/strong-oracle/disable.sql [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+DECLARE
+  c_action CONSTANT VARCHAR2(10) := 'DISABLE';
+BEGIN
```
Comment (MatthewPhinney) [inline src/sql/strong-oracle/disable.sql]: I'm not really qualified to review this, but it looks pretty straightforward. In any case, it's just used for testing, right?

Comment (clark800) [inline src/sql/strong-oracle/disable.sql]: Yeah, just for truncation in testing

**Comment by MatthewPhinney** [comment]
LGTM

File: src/sql/strong-oracle/disable.sql (hunk 4 above)
Comment (clark800) [inline src/sql/strong-oracle/disable.sql]: I think so, but let's do that in a future refactoring so we can merge this before my vacation
